AnalysisType: rule
DedupPeriodMinutes: 15
Description: >-
  A Duo authentication was denied due to an anomalous 2FA push.
DisplayName: "Duo User Auth Denied For Anomalous Push"
Enabled: true
Filename: duo_user_anomalous_push.py
LogTypes:
  - Duo.Authentication
Reference: https://duo.com/docs/adminapi#authentication-logs
RuleID: "DUO.User.Denied.AnomalousPush"
Runbook: >-
  Follow up with the user to confirm they intended several pushes in quick succession.
Severity: Medium
Tags:
  - Duo
Tests:
  - ExpectedResult: true
    Log:
      "access_device":
        "ip": "12.12.112.25"
        "os": "Mac OS X"
      "application":
        "key": "D12345"
        "name": "Slack"
      "auth_device":
        "ip": "12.12.12.12"
      "event_type": "authentication"
      "factor": "duo_push"
      "reason": "anomalous_push"
      "result": "denied"
      "user":
        "name": "example@example.io"
    Name: anomalous_push_occurred
  - ExpectedResult: false
    Log:
      "access_device":
        "ip": "12.12.112.25"
        "os": "Mac OS X"
      "application":
        "key": "D12345"
        "name": "Slack"
      "auth_device":
        "ip": "12.12.12.12"
      "event_type": "authentication"
      "factor": "duo_push"
      "reason": "user_approved"
      "result": "success"
      "user":
        "name": "example@example.io"
    Name: good_auth
  - ExpectedResult: false
    Log:
      "access_device":
        "ip": "12.12.112.25"
        "os": "Mac OS X"
      "application":
        "key": "D12345"
        "name": "Slack"
      "auth_device":
        "ip": "12.12.12.12"
      "event_type": "authentication"
      "factor": "duo_push"
      "reason": "out_of_date"
      "result": "denied"
      "user":
        "name": "example@example.io"
    Name: denied_old_creds
